### ISSUE TITLE
TOPPView freezes when exporting to SVG

### ISSUE THREAD

**Issue by hendrikweisser** [OpenMS/OpenMS#5380]
This is a follow-up to one important aspect of #5323 (on Ubuntu 20.04):
> Intermittently, the "Save File" dialog will freeze up completely.

The freeze is triggered by trying to interact with the file type selection drop-down menu in the "Save" dialog. It doesn't freeze every time, but when it does, it's accompanied by the following error messages in the terminal (which then also becomes unusable):
```
(TOPPView:2612283): Gdk-CRITICAL **: 13:27:05.308: gdk_window_get_window_type: assertion 'GDK_IS_WINDOW (window)' failed
(TOPPView:2612283): Gtk-CRITICAL **: 13:27:05.312: _gtk_widget_captured_event: assertion 'WIDGET_REALIZED_FOR_EVENT (widget, event)' failed
```
The second message is repeated whenever the mouse is moved over the "Save" dialog window, i.e. lots of times.

The following factors may increase to the likelihood of encountering the problem, though they don't seem strictly necessary:
1. Opening a larger mzML file (e.g. the BSA example files don't work well for reproducing the bug)
2. Using a larger TOPPView window or screen size
3. Using Identification view (via "Tools" > "Annotate with peptide identifications")
4. Moving peak labels in Identification view before trying to save the image

I've made a video showing a screen recording of the problem, where I open an mzML + idXML in "Identification view" and try to save an SVG image of an annotated spectrum. The first time (40 seconds in) it works, but then I move some labels around and the next attempt (~1 min. in) freezes TOPPView.

https://user-images.githubusercontent.com/2251068/122562005-71271780-d03a-11eb-8ea9-946c1a5fbdb9.mp4

**Comment by oliveralka** [comment]
Was not able to reproduce the issue on macOS 10.15.7, OpenMS develop (20210705).

@poshul Have you had time to take a look on Ubuntu?

**Comment by poshul** [comment]
@oliveralka I've tried it on both 21.04 and 18.04 LTS under WSL. I've not been able to reproduce the issue. @hendrikweisser does the issue occur with other desktop environments? I haven't tried it on Gnome Classic.

**Comment by hendrikweisser** [comment]
:-( Thanks for trying, @oliveralka and @poshul. Have you been able to move labels around in ID view and then save to SVG?

The desktop environment we use is MATE; I haven't tested any others. However, the "Gtk-CRITICAL" error messages lead me to believe that this is a Qt issue at heart.

**Comment by oliveralka** [comment]
> :-( Thanks for trying, @oliveralka and @poshul. Have you been able to move labels around in ID view and then save to SVG?

Yes, even with labels moved and waiting a minute or two - it worked without a problem on macOS.

**Comment by poshul** [comment]
There is an upstream GTK bug (https://gitlab.gnome.org/GNOME/gtk/-/issues/885) that I think might be related.

**Comment by hendrikweisser** [comment]
I now realise that I mentally conflated Gtk (from the error) and Qt, instead of thinking "Gtk -> GNOME". Oops!
So it's actually a very good suggestion to try a different desktop environment.

**Comment by cbielow** [comment]
while switching desktops is not really the best fix, is there anything we can do? It looks like the fix needs to happen upstream.

**Comment by timosachsenberg** [comment]
Closed as stale.
